src/client.rs

```diff
@@ -77,20 +77,26 @@ pub struct Client<S, T> {
     connected_to_server: bool,
 
     /// Name of the server pool for this client (This comes from the database name in the connection string)
-    target_pool_name: String,
+    pool_name: String,
 
     /// Postgres user for this client (This comes from the user in the connection string)
-    target_user_name: String,
+    username: String,
 
     /// Used to notify clients about an impending shutdown
-    shutdown_event_receiver: Receiver<()>,
+    shutdown: Receiver<()>,
+
+    /// Notify we're done.
+
+    #[allow(dead_code)]
+    drain: Sender<()>,
 }
 
 /// Client entrypoint.
 pub async fn client_entrypoint(
     mut stream: TcpStream,
     client_server_map: ClientServerMap,
-    shutdown_event_receiver: Receiver<()>,
+    shutdown: Receiver<()>,
+    drain: Sender<()>,
```

Comment:
Seems like drain isn't used at all here and we pass to all these functions just to implicitly drop it at the end. We could instead have an explicit drop at the end of the tokio task which is spawned for client entrypoint. This way we can remove unnecessary parameters in these functions